src/nupic/algorithms/Cells4.cpp

```diff
@@ -1036,11 +1029,12 @@ void Cells4::updateInferenceState(const std::vector<UInt> & activeColumns)
   _infActiveStateT1 = _infActiveStateT;
   _infPredictedStateT1 = _infPredictedStateT;
   memcpy(_cellConfidenceT1, _cellConfidenceT, _nCells * sizeof(_cellConfidenceT[0]));
+  memset(_cellConfidenceT, 0, _nCells * sizeof(_cellConfidenceT[0]));
```

Comment:
Is this (and the memset later) necessary or redundant? If all of the values are set then no need to zero them first. I haven't looked through the code thoroughly yet but it seems like it would be a pretty obvious bug if the values should have been zeroed but weren't.